contracts/FlowIDTableStaking.cdc

```diff
@@ -903,59 +953,105 @@ pub contract FlowIDTableStaking {
         pub fun removeUnapprovedNodes(approvedNodeIDs: {String: Bool}) {
             let allNodeIDs = FlowIDTableStaking.getNodeIDs()
 
-            // Reset the candidate node list
-            // TODO: Will be used for random slot selection in a later update
-            let candidateNodes = FlowIDTableStaking.account.load<{UInt8: {String: Bool}}>(from: /storage/idTableCandidateNodes) ?? {}
-            let emptyCandidateNodes: {UInt8: {String: Bool}} = {1: {}, 2: {}, 3: {}, 4: {}, 5: {}}
-            FlowIDTableStaking.account.save(emptyCandidateNodes, to: /storage/idTableCandidateNodes)
-
             for nodeID in allNodeIDs {
                 let nodeRecord = FlowIDTableStaking.borrowNodeRecord(nodeID)
 
                 let totalTokensCommitted = nodeRecord.nodeFullCommittedBalance()
 
-                // Remove nodes if they do not meet the minimum staking requirements
-                if !FlowIDTableStaking.isGreaterThanMinimumForRole(numTokens: totalTokensCommitted, role: nodeRecord.role) ||
-                   (approvedNodeIDs[nodeID] == nil) {
+                let greaterThanMin = FlowIDTableStaking.isGreaterThanMinimumForRole(numTokens: totalTokensCommitted, role: nodeRecord.role)
+                let nodeIsApproved: Bool =  approvedNodeIDs[nodeID] != nil
```

Comment:
I am consfused here, if there is a value in approvedNodeIds for the corresponding `nodeID` then it would be true?

Reply:
yeah, if the value for the node is non-nil, then that means it is approved